hack/generator/pkg/codegen/testdata/ArmResource/Arm_oneof_resource_conversion_on_arm_type_only_azure.golden

```diff
@@ -243,8 +243,8 @@ func (fakeResourceSpec *FakeResource_Spec) ConvertToARM(name string, resolvedRef
 }
 
 // CreateEmptyARMValue returns an empty ARM value suitable for deserializing into
-func (fakeResourceSpec *FakeResource_Spec) CreateEmptyARMValue() interface{} {
-	return FakeResource_SpecARM{}
+func (fakeResourceSpec *FakeResource_Spec) CreateEmptyARMValue() genruntime.ARMResourceStatus {
```

Comment:
Like here, isn't the fact that the `_Spec` resource is impl-ing `ARMResourceStatus` sorta weird? It's not a `Status` right? related to my comment above about how maybe we need to rename this interface since possibly it's not actually representing "ARM Status's" anymore, and is really representing... any ARM type?

Reply:
Also, isn't there no practical different between `genruntime.ARMResourceStatus` and `interface{}` currently? Since `ARMResourceStatus` has no members? Probably some will get added in the future though (possibly in another one of your PRs?) What members are you adding? These non-status types will need them too obviously, so that they impl the interface

Reply:
Relates back to my WIP branch and my comments above.